test/old_tests/UnitTests/Composable.cpp

```diff
@@ -167,4 +169,120 @@ TEST_CASE("Composable conversions")
+            {
+            }
+
+            HRESULT __stdcall ToString(HSTRING* result) noexcept final
+            {
+                return WindowsDuplicateString(static_cast<HSTRING>(get_abi(m_value)), result);
+            }
+
+            HRESULT __stdcall QueryInterface(GUID const& id, void** result) noexcept final
+            {
+                if (is_guid_of<IStringable>(id))
+                {
+                    *result = static_cast<ABI::Windows::Foundation::IStringable*>(this);
+                    AddRef();
+                    return S_OK;
+                }
+
+                return m_object->QueryInterface(id, result);
+            }
+
+            ULONG __stdcall AddRef() noexcept final
+            {
+                return 1 + m_references.fetch_add(1, std::memory_order_relaxed);
+            }
+
+            ULONG __stdcall Release() noexcept final
+            {
+                uint32_t const remaining = m_references.fetch_sub(1, std::memory_order_relaxed) - 1;
+
+                if (remaining == 0)
+                {
+                    delete this;
+                }
+
+                return remaining;
+            }
+
+            HRESULT __stdcall GetIids(ULONG* count, GUID** iids) noexcept final
+            {
+                return m_object->GetIids(count, iids);
+            }
+
+            HRESULT __stdcall GetRuntimeClassName(HSTRING* result) noexcept final
+            {
+                return m_object->GetRuntimeClassName(result);
+            }
+
+            HRESULT __stdcall GetTrustLevel(::TrustLevel* result) noexcept final
+            {
+                return m_object->GetTrustLevel(result);
+            }
+
+        private:
+
+            com_ptr<::IInspectable> m_object;
+            hstring m_value;
+            std::atomic<uint32_t> m_references{ 1 };
+        };
+
+        *result = new (std::nothrow) stringable(object, value);
+        return *result ? S_OK : E_OUTOFMEMORY;
+    }
+}
+
+TEST_CASE("Composable tearoff")
+{
+    static std::array<winrt::guid, 1> tearoff_iids{ winrt::guid_of<IStringable>() };
+
+    struct Tearoff : DerivedT<Tearoff, IClosable>
+    {
+        void Close()
+        {
+        }
+
+        int32_t query_interface_tearoff(winrt::guid const& id, void** result) const noexcept final
+        {
+            if (is_guid_of<IStringable>(id))
+            {
+                return make_stringable(*this, L"ToString", result);
+            }
+
+            *result = nullptr;
+            return E_NOINTERFACE;
+        }
+
+        winrt::array_view<winrt::guid> get_iids_tearoff() const noexcept final
+        {
+            return winrt::array_view(tearoff_iids);
+        }
+    };
+
+	auto object = make<Tearoff>();
+    auto ifaces = get_interfaces(object);
+
+    REQUIRE(object.as<IClosable>());
+    REQUIRE(object.as<IStringable>());
+	// IBaseOverrides is repeated twice for some reason, so actual size is 7 but there are only 6 unique interfaces
```

Comment:
This seems odd and possibly worth understanding before proceeding.  Would you be willing to check the behavior of this test case with your `get_iid_tearoff` logic commented out?

Reply:
It doesn't come from `get_iid_tearoff `. It is likely due to this being a composable type and the `get_local_iids()` of the outer class also picks up the overrideable interfaces from the inner class since overrideable interfaces get special handling.